Davi_Cruz/variable.cpp

```diff
@@ -0,0 +1,43 @@
+#include "variable.h"
+
+double Table_variables::define_variable(string variable_name, double variable_value)
+{
+    if (is_defined(variable_name))
+    {
+        error(variable_name,"declared twice.");
+    }
+    variables.push_back({Variable{variable_name, variable_value}});
+    return variable_value;
+}
+
+double Table_variables::get_value(string variable_name)
+{
+    for (auto &&i : variables)
```

Comment:
Why do you use auto&& ?

Reply:
Vs code uses it for interaction with vectors. Only keyboard shortcut.